### ISSUE TITLE
.NET 3.5 version can't create secure channel

### ISSUE THREAD

**Issue by PhilPJL** [newman55/unity-mod-manager#77]
Running the .NET 3.5 release version 0.23.3 on Windows 10 shows an error.

![image](https://user-images.githubusercontent.com/271656/113841921-1fd0f100-978a-11eb-9288-eb441ba96997.png)

I assume this is related to this line in Form.cs.
`ServicePointManager.SecurityProtocol = SecurityProtocolType.Ssl3 | SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;`

Since `SecurityProtocolType.Tls11` and `SecurityProtocolType.Tls12` are .NET 4.x only.

I'm unable to build the .NET 3.5 version of the main branch for some reason, so can't verify.

**Comment by newman55** [comment]
You need to choose the Release-net35 configuration.

**Comment by PhilPJL** [comment]
Something's a bit screwy.  If I download the repo from fresh, and build debug, release and release-net35 I get an empty release-net35 folder.  There is no UnityModManager-net35.exe.

If I have Release-net35 selected NET35 doesn't appear to be defined.

**Comment by PhilPJL** [comment]
If I set the UMM app to .NET 3.5 and set the NET35 compilation symbol I can run the .NET 3.5 version.  Then I get errors due to the ISupportInitialize issue, so these
        //    ((System.ComponentModel.ISupportInitialize)(this.splitContainerMain)).EndInit();
have to be commented out.
The app also fails to start because the app.config is referencing System.Configuration Version=4.0.0, whereas it needs Version 2.0.0.
Fixing that as well allows the app to run, but then I get errors when it tries to detect updates, which may be due to an issue running on Windows 10.

If I revert the changes and rebuild again for .net 4.5 everything is ok. 
Is a .NET 3.5 version really required?

**Comment by PhilPJL** [comment]
I've reported this to MS https://developercommunity.visualstudio.com/t/winforms-designer-issue-with-isupportinitialize/1392601?from=email

**Comment by newman55** [comment]
I don't understand why it works in the main branch, but it doesn't work here.

**Comment by newman55** [comment]
You can decompile UnityModManager-net35.exe and see exactly the same code there.
```
((System.ComponentModel.ISupportInitialize)(this.splitContainerMain)).BeginInit();
```

**Comment by PhilPJL** [comment]
Yes I've done that, it's strange.  Are you using VS 2019?

**Comment by newman55** [comment]
2017

**Comment by PhilPJL** [comment]
Ok that will be the issue I guess with the .NET 3.5 build for me using 2019.
I may have to install 2017 just to check.

Can you replicate the 'the request was aborted' issue above?

**Comment by PhilPJL** [comment]
Or you can install 2019?

**Comment by newman55** [comment]
i have

**Comment by PhilPJL** [comment]
The UMM dll has to be .NET 3.5, but do you really need a .NET 3.5 app?  I also note that the Updater app is .NET 4.5.  Shouldn't that be .NET 3.5 also if the main app is .NET 3.5?

**Comment by newman55** [comment]
I had issue that the net45 version does not work for everyone, net35 is needed as an alternative. The updater is not important.

**Comment by PhilPJL** [comment]
I've downloaded VS 2017 community edition.  Same issue with the .NET 3.5 build.  Doesn't happen.

**Comment by PhilPJL** [comment]
The fix for the network error is 
![image](https://user-images.githubusercontent.com/271656/114005844-da7af500-9857-11eb-8125-da823afe762c.png)
from
https://www.aspsnippets.com/Articles/Net-WebClient-Could-not-create-SSLTLS-secure-channel.aspx
The fix may be needed elsewhere.

Note that to test this I have to set the debug configuration to .NET 3.5 and add the NET35 symbol, fix the designer code  to remove ISupportInitialize and change the app.config to use Version=2.0.0.0
![image](https://user-images.githubusercontent.com/271656/114006112-11510b00-9858-11eb-9d33-882892266b88.png)

**Comment by newman55** [comment]
The fact is that with any change in the designer, the Form.Designer.cs file is generated entirely.

**Comment by PhilPJL** [comment]
Yes that's right.  That's what I've raised to MS.  The designer, at least in VS2019, when modifying a multi-targeted app should emit code which works correctly on all platforms.

**Comment by PhilPJL** [comment]
Fixed in 0.23.4
